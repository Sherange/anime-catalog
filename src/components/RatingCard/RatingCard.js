import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {primaryColor} from '../../constans/theme';
import PropTypes from 'prop-types';

const RatingCard = props => {
  const {title, rate} = props;
  return (
    <View style={styles.ratingCard}>
      <Text>{title}</Text>
      <Text style={styles.ratingCardText}>{rate}</Text>
    </View>
  );
};

RatingCard.propTypes = {
  title: PropTypes.string.isRequired,
  rate: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

const styles = StyleSheet.create({
  ratingCard: {
    width: 100,
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    backgroundColor: '#FFFFFF',
    shadowColor: '#000000',
    shadowOpacity: 0.2,
    shadowRadius: 1,
    shadowOffset: {
      height: 1,
      width: 1,
    },
  },
  ratingCardText: {
    fontSize: 14,
    marginTop: 8,
    fontWeight: '600',
    color: primaryColor,
  },
});

export default RatingCard;
