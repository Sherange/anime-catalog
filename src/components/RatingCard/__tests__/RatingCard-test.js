import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RatingCard from '../RatingCard';

it('RatingCard renders correctly', () => {
  renderer.create(<RatingCard title="Rank" rate={10} />);
});

test('RatingCard snapshot', () => {
  const snapshot = renderer
    .create(<RatingCard title="Rank" rate={10} />)
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});
