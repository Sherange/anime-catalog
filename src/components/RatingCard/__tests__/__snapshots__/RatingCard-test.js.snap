// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingCard snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 20,
      "height": 60,
      "justifyContent": "center",
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 1,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1,
      "width": 100,
    }
  }
>
  <Text>
    Rank
  </Text>
  <Text
    style={
      Object {
        "color": "#D9251D",
        "fontSize": 14,
        "fontWeight": "600",
        "marginTop": 8,
      }
    }
  >
    10
  </Text>
</View>
`;
